src/tests/file-sys-add-fixed-map-api-test/file-sys-add-fixed-map-api-test.c

```diff
@@ -0,0 +1,78 @@
+/*
+ *      Copyright (c) 2020, United States government as represented by the
+ *      administrator of the National Aeronautics Space Administration.
+ *      All rights reserved. This software was created at NASA Goddard
+ *      Space Flight Center pursuant to government contracts.
+ *
+ *      This is governed by the NASA Open Source Agreement and may be used,
+ *      distributed and modified only according to the terms of that agreement.
+ */
+
+/*
+ * Filename: file-sys-add-fixed-map-api-test.c
+ *
+ * Purpose: This file contains functional tests for "osapi-FileSysAddFixedMap"
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "common_types.h"
+#include "osapi.h"
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+
+
+
+/* *************************************** MAIN ************************************** */
+
+void TestFileSysAddFixedMapApi(void)
+{
+    int32 expected;
+    int32 actual;
+    uint32 fs_id;
+
+    /* Test for nominal inputs */
+    /*
+     * This test case requires a fixed virtual dir for one test case.
+     * Just map /test to a dir of the same name, relative to current dir.
+     */
+
+    expected = OS_SUCCESS;
+    actual = OS_FileSysAddFixedMap(&fs_id, "./test", "/test");
+    UtAssert_True(actual == expected, "OS_FileSysAddFixedMap() (%ld) == OS_SUCCESS", (long)actual);
```

Comment:
Is there an action you can perform to make sure this worked?  Open a file in on directory and make sure it shows up in the map or similar?

Reply:
Note that this function is also relied upon by some of existing unit tests (file-api-test, osloader-test) which first add a fixed map using this function and then proceed to open/close files within that directory.  So this is already covered to some degree.

If you really want to do something here to confirm it worked in this test, one other option (which I just thought of) is using the `OS_TranslatePath` API.  We do expose this in the public API for some reason.

If one tries to translate `/test/myfile.txt` before adding this map, it should fail, but after adding this map it should succeed and return `./test/myfile.txt`.